core/src/proxy/block_engine_stage.rs

```diff
@@ -102,28 +108,92 @@ impl BlockEngineStage {
+                let exit = exit.clone();
+                let block_builder_fee_info = block_builder_fee_info.clone();
+
+                move || {
+                    let rt = tokio::runtime::Builder::new_multi_thread()
+                        .enable_all()
+                        .build()
+                        .unwrap();
+                    rt.block_on(Self::start(
+                        block_engine_config,
+                        cluster_info,
+                        bundle_tx,
+                        packet_tx,
+                        banking_packet_sender,
+                        exit,
+                        block_builder_fee_info,
+                        true, // is_primary
+                    ));
+                }
             })
             .unwrap();
+        handles.push(primary_handle);
+
+        // Secondary engine threads
+        for (index, url) in secondary_urls.into_iter().enumerate() {
+            if !url.is_empty() {
+                let secondary_handle = Builder::new()
+                    .name(format!("block-engine-secondary-{}", index))
+                    .spawn({
+                        let cluster_info = cluster_info.clone();
+                        let bundle_tx = bundle_tx.clone();
+                        let packet_tx = packet_tx.clone();
+                        let banking_packet_sender = banking_packet_sender.clone();
+                        let exit = exit.clone();
+                        let block_builder_fee_info = block_builder_fee_info.clone();
+                        
+                        move || {
+                            let rt = tokio::runtime::Builder::new_multi_thread()
+                                .enable_all()
+                                .build()
+                                .unwrap();
+                            
+                            // Create immutable config for secondary engine
+                            let secondary_config = Arc::new(Mutex::new(BlockEngineConfig {
+                                block_engine_url: url,
+                                secondary_block_engine_urls: Vec::new(),
+                                trust_packets: false, // Always false for secondary
+                            }));
+                            
+                            rt.block_on(Self::start(
+                                secondary_config,
+                                cluster_info,
+                                bundle_tx,
+                                packet_tx,
+                                banking_packet_sender,
+                                exit,
+                                block_builder_fee_info, // Share fee info instance but won't update it
+                                false, // is_primary
+                            ));
+                        }
+                    })
+                    .unwrap();
+                handles.push(secondary_handle);
+            }
```

Comment:
This is a big duplication of the code above, is it possible to just move this spawn into a function that we parameterize by passing in `config: Either<PrimaryConfig, SecondaryConfig>`